### PR TITLE
Add support for decrypting a version-prefixed ciphertext

### PR DESCRIPTION
**Pull request by atheriel**
Some existing implementations prefix the encrypted ciphertext with a version byte (which is always `0x01` to my knowledge). This commit adds support for decrypting these payloads.

This change includes an internal API to generate versioned ciphertext for testing, but it does not add a public one. This could be done in the future simply by exposing the existing private method.

### CODE REVIEW

File: crypt/key.go [hunk 1]
```diff
@@ -123,6 +141,31 @@ func (k *Key) Decrypt(s string) (string, error) {
...
+	switch buf[0] {
+	case byte(1):
+		bytes, err := k.decryptSecretbox(buf[1:])
+		if err != nil {
+			// In order to handle the (unlikely but possible) case
+			// where a versionless payload *just happens* to start
+			// with byte(1), we must also try the fallback here.
+			goto fallback
+		}
+		return string(bytes[:]), nil
```
Comment (jonyoder) [inline crypt/key.go:157]: Instead of using a `goto`, you could simply invert the conditional and return here `if err == nil`.

**Review by jonyoder — COMMENTED** [review]
> Some existing implementations prefix the encrypted ciphertext with a version byte

Can you explain this statement a bit more? Specifically, what are the existing implementations and what is the purpose of the version byte?

File: crypt/key.go (hunk 1 above)
Comment (atheriel) [reply crypt/key.go]: Ah, good catch. Fixed.

**Review by jonyoder — APPROVED** [review]

**Comment by atheriel** [comment]
I discussed the existing implementations with Jon directly.

**Pull request merged**